sympy/polys/matrices/domainmatrix.py

```diff
@@ -204,15 +462,43 @@ def lu_solve(self, rhs):
         return self.from_rep(sol)
 
     def charpoly(self):
+        """Computes characteristic polynomial of a square matrix
+
+        Examples
+        ========
+
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> from sympy import QQ
+        >>> A = DomainMatrix([
+        ...     [QQ(2), QQ(-1), QQ(0)],
+        ...     [QQ(-1), QQ(2), QQ(-1)],
+        ...     [QQ(0), QQ(0), QQ(2)]], (3, 3), QQ)
+        >>> A.charpoly()
+        [1, -6, 11, -6]
```

Comment:
It's best to use the simplest nontrivial example that you can.

It would be nicer to choose a 2x2 example like:
```
        >>> A = DomainMatrix([[ZZ(1), ZZ(2)]
        ...                   [ZZ(3), ZZ(4)]], (2, 2), ZZ)
```